kratos/utilities/auxiliar_model_part_utilities.h

```diff
@@ -0,0 +1,339 @@
+     * @param ThisIndex The index of the mesh where remove the entity
+     */
+    void RemoveConditionAndBelongings(Condition::Pointer pThisCondition, Flags IdentifierFlag = TO_ERASE, IndexType ThisIndex = 0);
+
+    /** 
+     * @brief Remove the condition with given Id from mesh with ThisIndex in parents, itself and children. This method removes belonging entities too
+     * @details The method check that when removing the nodes doesn't affect to other entities 
+     * @param ConditionId The ID of the condition to remove
+     * @param IdentifierFlag The flag that identifies the entities to remove
+     * @param ThisIndex The index of the mesh where remove the entity
+     */
+    void RemoveConditionAndBelongingsFromAllLevels(IndexType ConditionId, Flags IdentifierFlag = TO_ERASE, IndexType ThisIndex = 0);
+
+    /** 
+     * @brief Remove given condition from mesh with ThisIndex in parents, itself and children. This method removes belonging entities too
+     * @details The method check that when removing the nodes doesn't affect to other entities 
+     * @param rThisCondition The reference to the condition to remove
+     * @param IdentifierFlag The flag that identifies the entities to remove
+     * @param ThisIndex The index of the mesh where remove the entity
+     */
+    void RemoveConditionAndBelongingsFromAllLevels(Condition& rThisCondition, Flags IdentifierFlag = TO_ERASE, IndexType ThisIndex = 0);
+
+    /** 
+     * @brief Remove given condition from mesh with ThisIndex in parents, itself and children. This method removes belonging entities too
+     * @details The method check that when removing the nodes doesn't affect to other entities 
+     * @param pThisCondition The pointer to the condition to remove
+     * @param IdentifierFlag The flag that identifies the entities to remove
+     * @param ThisIndex The index of the mesh where remove the entity
+     */
+    void RemoveConditionAndBelongingsFromAllLevels(Condition::Pointer pThisCondition, Flags IdentifierFlag = TO_ERASE, IndexType ThisIndex = 0);
+
+    /** 
+     * @brief It erases all conditions identified by "IdentifierFlag" by removing the pointer. This method removes belonging entities too
+     * @details Pointers are erased from this level downwards nodes will be automatically destructured when no pointer is left to them
+     * The method check that when removing the nodes doesn't affect to other entities 
+     * @param IdentifierFlag The flag that identifies the entities to remove
+     */
+    void RemoveConditionsAndBelongings(Flags IdentifierFlag = TO_ERASE);
+
+    /** 
+     * @brief It erases all conditions identified by "IdentifierFlag" by removing the pointer. This method removes belonging entities too
+     * @details Pointers will be erase from all levels nodes will be automatically destructured when no pointer is left to them
+     * The method check that when removing the nodes doesn't affect to other entities 
```

Comment:
so what happens in case a node does belong to another entity? this shall be documented...

Reply:
It removes the other entities, check the code. If you remove the element, remove the conditions conatining that nodes and vice versa

Reply:
checking the code is exaclty what i don't want to do. 

I mean that If i read the docstring i should understand what this is doing.

Reply:
But this basically what is explained, maybe poorly, in the docstring. BTW comments can lie, but code can't

Reply:
"This method removes belonging entities too" this is in the docstring, maybe the problem is that I prefer to use the dynamic line construction of the IDEs (that's why I put all the comments in one line), because depending of the screen used it adapts. I will change it